File: android/MyPlaygroundInJava/app/src/main/java/com/gmail/htaihm/myplaygroundinjava/colorchanger/ColorChangerViewModel.java
package com.gmail.htaihm.myplaygroundinjava.colorchanger;

import androidx.annotation.ColorInt;
import androidx.lifecycle.ViewModel;

/** A simple {@link ViewModel} that stores a color resource */
public class ColorChangerViewModel extends ViewModel {

  private int mColorResource = 0xfff;

  public void setColorResource(@ColorInt int colorResource) {
    mColorResource = colorResource;
  }

  @ColorInt
  public int getColorResource() {
    return mColorResource;
  }
}

File: android/jetpack/ColorChanger/app/src/main/java/com/gmail/htaihm/colorchanger/ColorChangerViewModel.kt
package com.gmail.htaihm.colorchanger

import android.arch.lifecycle.ViewModel

class ColorChangerViewModel : ViewModel() {
    private var colorResource: Int = 0xfff
    fun setColorResource(colorResource: Int) {
        this.colorResource = colorResource
    }
    fun getColorResource() = colorResource
}